lib/Optimizer/Transforms/LiftArrayAlloc.cpp

```diff
@@ -65,10 +68,11 @@ class AllocaPattern : public OpRewritePattern<cudaq::cc::AllocaOp> {
     Value conArr;
     Value conGlobal;
     if (toGlobal) {
-      static unsigned counter = 0;
+      // static unsigned counter = 0;
       auto ptrTy = cudaq::cc::PointerType::get(arrTy);
       // Build a new name based on the kernel name.
-      std::string name = funcName + ".rodata_" + std::to_string(counter++);
+      /// FIXME: Add a unique identifier per set of parameters
```

Comment:
Any parameters, for the transformation to proceed, will be indistinguishable from other constant data.